Tools/WrapperGenerator/Templates/TypeScript/Partials/Method/TaskMethod.mustache

```diff
@@ -61,24 +68,20 @@ async {{ name }}(
             return;
         }
 {{/ type.is_result_base }}
-{{/ results }}
 
-        _resolve(
-{{# results }}
-{{# type.is_class_or_interface }}
-            _{{ name }}Instance{{> Comma }}
-{{/ type.is_class_or_interface }}
-{{^ type.is_class_or_interface }}
-{{# type.is_string }}
-            Module.UTF8ToString({{name}}){{> Comma }}
-{{/ type.is_string }}
-{{^ type.is_string }}
-            {{# type.is_bool}}!!{{/ type.is_bool }}{{ name }}{{> Comma }}
-{{/ type.is_string }}
-{{/ type.is_class_or_interface }}
-{{/ results }}
-        )
+{{# type.is_result_base }}
+    if ({{> ResultName }}.getResultCode() == Services.EResultCode.Success) {
+        
+{{/ type.is_result_base }}
+        _resolve({{> ResultName }})
+        
+{{# type.is_result_base }}
+    } else {
+        _reject(new Error({{> ResultName }}.getResponseBody()));
```

Comment:
This does not return enough information to the clients. We expose not only the response body, but the HTTP error code and the parsed X-ErrorCode value. This also prevents result objects from being manually disposed as they are not returned to the caller.

Reply:
Ideally what we need here is an simple javascript interface for Errors.  Ideally with an agreed error code type and message.  http code should be redundant and not exposed to the client (as it is internal implementation). The error code should cover the type of error already.
{
errorCode: 3,
message: "Resource not found"
}